crates/kitsune_p2p/kitsune_p2p/src/gossip/simple_bloom/step_4_com_loop_inner.rs

```diff
@@ -0,0 +1,337 @@
+                                    key.clone(),
+                                    data.clone(),
+                                ));
+                            }
+                            MetaOpData::Agent(agent_info_signed) => {
+                                // TODO - we actually only need to do this
+                                // once, since the agent store is shared...
+                                futs.push(i.evt_sender.put_agent_info_signed(
+                                    PutAgentInfoSignedEvt {
+                                        space: i.space.clone(),
+                                        agent: agent.clone(),
+                                        agent_info_signed: agent_info_signed.clone(),
+                                    },
+                                ));
+                            }
+                        }
+                    }
+                    let key = chunk.key();
+                    i.local_bloom.set(&key);
+                    i.local_data_map.insert(key, chunk);
+                }
+
+                Ok(futs)
+            })?;
+
+            if !futs.is_empty() {
+                futures::future::try_join_all(futs)
+                    .await
+                    .map_err(KitsuneError::other)?;
+            }
+
+            return Ok(());
+        }
+    };
+
+    let remote_filter = decode_bloom_filter(&remote_filter);
+
+    // send accept if applicable / gather the keys the remote needs
+    let con_clone = con.clone();
+    let out_keys = inner.share_mut(move |i, _| {
+        // for now, just always accept gossip initiates
+        if send_accept {
+            let local_filter = encode_bloom_filter(&i.local_bloom);
+            let gossip = GossipWire::accept(local_filter);
+            let peer_cert = con_clone.peer_cert();
+            i.outgoing
+                .push((peer_cert, HowToConnect::Con(con_clone), gossip));
+        }
+
+        let mut out_keys = Vec::new();
+
+        // find the keys for data the remote doesn't have
+        for key in i.local_key_set.iter() {
+            if !remote_filter.check(key) {
+                out_keys.push(key.clone());
+            }
+        }
+
+        let local_op_count = i.local_key_set.len();
+        let remote_needs_op_count = out_keys.len();
+        tracing::debug!(%local_op_count, %remote_needs_op_count, "identified ops remote needs");
+
+        Ok(out_keys)
+    })?;
+
+    if out_keys.is_empty() {
+        // the remote doesn't need anything from us
+        // ... if we initiated this gossip, mark it as done.
+        inner.share_mut(move |i, _| {
+            if let Some(tgt_cert) = i.initiate_tgt.clone() {
+                if con.peer_cert() == tgt_cert {
+                    i.initiate_tgt = None;
+                }
+            }
+
+            // publish an empty chunk incase it was the remote who initiated
+            let gossip = GossipWire::chunk(true, Vec::new());
```

Comment:
```suggestion
            let gossip = GossipWire::chunk(true, Vec::with_capacity(0));
```

Reply:
Actually, I was reading the source code for Vec the other day, they always start with zero capacity, so this isn't needed.